helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java

```diff
@@ -0,0 +1,76 @@
+package org.apache.helix.rest.server.resources.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Collections;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+import org.apache.helix.AccessOption;
+import org.apache.helix.PropertyPathBuilder;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZkBaseDataAccessor;
+import org.apache.helix.store.HelixPropertyStore;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@Path("/clusters/{clusterId}/propertyStore")
+public class PropertyStoreAccessor extends AbstractHelixResource {
+  private static Logger LOG = LoggerFactory.getLogger(PropertyStoreAccessor.class);
+
+  @GET
+  @Path("{path: .+}")
+  public Response getPropertyByPath(@PathParam("clusterId") String clusterId,
+      @PathParam("path") String path) {
+    path = "/" + path;
+    if (!isPathValid(path)) {
+      LOG.error("The path {} is mis-inputted for cluster {}", path, clusterId);
+      throw new WebApplicationException(Response.status(Response.Status.BAD_REQUEST).entity(
+          "Invalid path string. Valid path strings use slash as the directory separator and names the location of ZNode")
+          .build());
+    }
+    String propertyStoreRootPath = PropertyPathBuilder.propertyStore(clusterId);
+    HelixPropertyStore<ZNRecord> propertyStore = new ZkHelixPropertyStore<>(
+        (ZkBaseDataAccessor<ZNRecord>) getDataAccssor(clusterId).getBaseDataAccessor(),
+        propertyStoreRootPath, Collections.emptyList());
+    ZNRecord record = propertyStore.get(path, null, AccessOption.PERSISTENT);
```

Comment:
Here's a concern:

What if the ZNode under a PropertyStore is not a valid ZNRecord ZNode? In that case, this will throw an exception.

I think the idea that we discussed was to either:
1) return an appropriate error response saying the data is invalid. I am leaning towards a 500.
2) Or return the ZNode binary (byte[]) in a ZNRecord format. The JSON would look like this:

{
  simpleField:
   "ZNODE_NAME" : "xxxxxx..."
}

What do you think? 

@dasahcc @jiajunwang @lei-xia Thoughts?

Reply:
How about returning the raw data (`byte[]`) and let the clients parse the content. Helix-rest only supports raw read/write APIs and don't make any assumption about the underlying data structure.

Reply:
@i3wangyi  That might be problematic. Many users of Helix REST already assume the JSON format as the format of the return value. And as a matter of fact, that is the established protocol for Helix REST endpoints. The format I suggested above

```
{
simpleField:
"ZNODE_NAME" : "xxxxxx..."
}
```
is a flexible format that meets all requirements.

Reply:
It's fair. The problem with our current accessors is they all assume ZNRecord property as the returned value. I've updated the PR using a pure ZKClient to retrieve propertyStore data. 
Please take a quick look and see if it's acceptable behavior. In the meantime, I'm figuring out how to add proper integration test to validate non-znRecord data. @narendly